### PR TITLE
Documentation for https://github.com/silverstripe/silverstripe-upgrader/

### PR DESCRIPTION
**Pull request by tractorcow**
See https://github.com/silverstripe/silverstripe-framework/issues/4559

### CODE REVIEW

File: docs/en/04_Changelogs/4.0.0.md [hunk 1]
```diff
@@ -302,6 +302,26 @@ E.g.
 
 ## Upgrading
 
+### General upgrading
+
+In order to better support migration of legacy modules and code, an automatic class upgrade tool
```
Comment (hafriedlander) [inline docs/en/04_Changelogs/4.0.0.md]: I don't think this gets to the USP quickly enough. People won't really understand what this tool is without context.

Maybe something like:

4.0 moves many classes to namespaces, and renames many methods. A tool is available to (semi-)automatically update your 3.x code.

**Comment by sminnee** [comment]
It's difficult to test this until we have some upgrade rules added to the framework. Let's get a simple case in and then confirm that these instructions work, before merging them.

**Comment by hafriedlander** [comment]
@sminnee the problem with doing that is that as soon as there are some rules in framework, it means framework is breaking for everyone that hasn't run the upgrader. So I'd rather have the docs in first. But I'll managing merging so there's a PR for the ORM namespacing that these docs can be tested against.

**Comment by tractorcow** [comment]
Rebased to resolve tests fixed upstream.

**Pull request merged**